src/handlemicrophone.js

```diff
@@ -54,10 +63,18 @@ exports.handleMicrophone = function(token, model, mic, callback) {
     'max_alternatives': 3,
     'inactivity_timeout': 600,
     'word_alternatives_threshold': 0.001,
-    'keywords_threshold': keywords_threshold,
-    'keywords': keywords,
-    'smart_formatting': true
+    'smart_formatting': true,
   };
+  
+  var keywords = display.getKeywordsToSearch();
+  if(keywords.length > 0) {
+    var keywords_threshold = 0.01;
+    options.message.keywords_threshold = keywords_threshold;
+    options.message.keywords = keywords;
+  }
+  var speaker_labels = $('li.speakersTab').is(':visible');
+  options.message.speaker_labels = speaker_labels;
```

Comment:
Agree, checking the value of the checkbox is preferable. On the other hand, li.speakersTab is visible iff the checkbox is checked.